retry.go

```diff
@@ -229,6 +231,10 @@ func (r *retryHTTP) execute() (res *http.Response, err error) {
 		for k, v := range r.headers {
 			req.Header.Set(k, v)
 		}
+		urlStrLen := len(r.fullURL.String())
+		if urlStrLen >= len(testPath) && r.fullURL.String()[urlStrLen-len(testPath):] == testPath {
```

Comment:
this is a bit janky, what are alternative ways that do not involve hardcoding a `test path` sentinel value in the production code path?

Reply:
I agree, but I couldn't think of a proper GET request that would just bypass this. Any ideas?

Reply:
Would http round tripper fix this? If we defined our own dummy roundtripper that just did nothing and returned no error